shell/platform/linux/fl_platform_views_plugin.cc

```diff
@@ -0,0 +1,349 @@
+  }
+
+  fl_platform_view_set_direction(platform_view, direction);
+
+  return FL_METHOD_RESPONSE(fl_method_success_response_new(nullptr));
+}
+
+// Mark that a mouse pointer has entered into the platform view.
+static FlMethodResponse* platform_views_enter(FlPlatformViewsPlugin* self,
+                                              FlValue* args) {
+  if (fl_value_get_type(args) != FL_VALUE_TYPE_LIST ||
+      fl_value_get_length(args) != 1) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Argument list missing or malformed", nullptr));
+  }
+
+  int64_t view_id = fl_value_get_int(fl_value_get_list_value(args, 0));
+  FlPlatformView* platform_view =
+      fl_platform_views_plugin_get_platform_view(self, view_id);
+  if (!platform_view) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Platform view not found", nullptr));
+  }
+
+  GtkWidget* widget = fl_platform_view_get_view(platform_view);
+  if (widget)
+    fl_gesture_helper_enter(self->gesture_helper, widget);
+
+  return FL_METHOD_RESPONSE(fl_method_success_response_new(nullptr));
+}
+
+// Mark that a mouse pointer has exited from the platform view.
+static FlMethodResponse* platform_views_exit(FlPlatformViewsPlugin* self,
+                                             FlValue* args) {
+  if (fl_value_get_type(args) != FL_VALUE_TYPE_LIST ||
+      fl_value_get_length(args) != 1) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Argument list missing or malformed", nullptr));
+  }
+
+  int64_t view_id = fl_value_get_int(fl_value_get_list_value(args, 0));
+  FlPlatformView* platform_view =
+      fl_platform_views_plugin_get_platform_view(self, view_id);
+  if (!platform_view) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Platform view not found", nullptr));
+  }
+
+  GtkWidget* widget = fl_platform_view_get_view(platform_view);
+  if (widget)
+    fl_gesture_helper_exit(self->gesture_helper, widget);
+
+  return FL_METHOD_RESPONSE(fl_method_success_response_new(nullptr));
+}
+
+// Disposes the platform view.
+static FlMethodResponse* platform_views_dispose(FlPlatformViewsPlugin* self,
+                                                FlValue* args) {
+  if (fl_value_get_type(args) != FL_VALUE_TYPE_INT) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Argument map missing or malformed", nullptr));
+  }
+
+  int64_t id = fl_value_get_int(args);
+
+  if (!g_hash_table_remove(self->platform_views, GINT_TO_POINTER(id))) {
+    return FL_METHOD_RESPONSE(fl_method_error_response_new(
+        kBadArgumentsError, "Platform view does not exist", nullptr));
+  }
+
+  return FL_METHOD_RESPONSE(fl_method_success_response_new(nullptr));
+}
+
+// Called when a method call is received from Flutter.
+static void method_call_cb(FlMethodChannel* channel,
```

Comment:
nit: rename to `method_call_handler`

Reply:
`method_call_cb`  in other place:
https://github.com/flutter/engine/blob/13729af4fbae6e220287c159a2cb2188fc805bbc/shell/platform/linux/fl_mouse_cursor_plugin.cc#L122

Reply:
Yeah, that's the convention we've used elsewhere.